### PR TITLE
fixes look controls in chrome (Fixes #983)

### LINKED ISSUE #983: Look controls not working in Chrome (dev branch)
Cannot read property 'addEventListener' of undefined error using Chrome (dev)

image

### PR DESCRIPTION
**Pull request by caseyyee**
Fixes https://github.com/aframevr/aframe/issues/983

### CODE REVIEW

File: src/components/look-controls.js [hunk 1]
```diff
@@ -28,7 +28,11 @@ module.exports.Component = registerComponent('look-controls', {
 
   play: function () {
     this.previousPosition.set(0, 0, 0);
-    this.addEventListeners();
+
+    // defer adding listeners until everything has run.
```
Comment (ngokevin) [inline src/components/look-controls.js]: what are we waiting on to run specifically? maybe we should use an event?

Comment (caseyyee) [inline src/components/look-controls.js]: `this.el.sceneEl.canvas` is undefined.    The scene canvas isn't setup before the event listener runs.

I tried using `loaded` event, but for some reason it runs multiple times on scene.  Not sure why this is.

Comment (ngokevin) [inline src/components/look-controls.js]: The `canvas` component will trigger `render-target-loaded` event on the scene. Want to try that?

Comment (caseyyee) [inline src/components/look-controls.js]: Same issue with the `Cannot read property 'render' of undefined` error in the console in the same issue.   

If I checked `scene.hasLoaded` it's ` false`.   Resolves to `true` in FF.   Just a difference in the order in which the methods are called In Chrome?

Comment (caseyyee) [inline src/components/look-controls.js]: I'll give that a shot.

**Comment by caseyyee** [comment]
@ngokevin ?

**Comment by ngokevin** [comment]
Okay, looks good. Gonna do some spot checks and wait for Travis.

File: src/core/scene/a-scene.js [hunk 2]
```diff
@@ -450,7 +450,9 @@ var AScene = module.exports = registerElement('a-scene', {
           });
         }
 
-        this.renderer.render(this.object3D, camera);
+        if (this.renderer) {
```
Comment (ngokevin) [inline src/core/scene/a-scene.js]: is there a place we can do this check earlier?

Comment (caseyyee) [inline src/core/scene/a-scene.js]: lemme take a look.

File: src/core/scene/a-scene.js [hunk 3]
```diff
@@ -375,9 +375,11 @@ var AScene = module.exports = registerElement('a-scene', {
           this.resize();
 
           // Kick off render loop.
-          this.render();
-          this.renderStarted = true;
-          this.emit('renderstart');
+          if (this.renderer) {
```
Comment (ngokevin) [inline src/core/scene/a-scene.js]: Might be worth filing an issue to see why renderer is not there at this point.

**Pull request merged**